### ISSUE TITLE
Restaurant demo flow is flaky.

### ISSUE THREAD

**Issue by shaper** [PolymerLabs/arcs-cdn#181]
Follow up from https://github.com/PolymerLabs/arcs-cdn/issues/178

I noted the restaurant demo flow still failed after the PR that should have fixed it landed:

https://travis-ci.org/PolymerLabs/arcs-cdn/builds/335769899

I can repro locally. It was passing 100% previously with my patch and is now failing 100% of the time. Sample error bits below. Requires further investigation.

Maybe the network request is not always succeeding. We've discussed mocking it for test purposes.

```
running web tests on version dev
click: allSuggestions
Trying to accept: Find restaurants
click: desiredSuggestion
Accepted suggestion: Find restaurants
click: restaurantSelectors
Trying to accept: Make a reservation
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
Couldn't find suggestion 'Make a reservation'.
```

**Comment by shaper** [comment]
The suggestions are showing only `undefined` some of the time:

![error_chrome_2018-01-31t20-03-00 180z](https://user-images.githubusercontent.com/8998/35644835-ec91ca48-067e-11e8-9fbe-4d4d2fca0408.png)

**Comment by shaper** [comment]
The `ArcHost` plans that produce `undefined` are logged after the one with `Find restaurants` is selected, as below.

Somehow `Make a reservation` description is lost now.

```
(5) [{…}, {…}, {…}, {…}, {…}, generations: Array(7)]
0:{plan: Recipe, rank: 0.04000000000000001, description: Description, descriptionText: "Show Claire's wishlist (Book: How to Draw plus 2 other items).", hash: "3715"}
1:{plan: Recipe, rank: 0.000002560000000000002, description: Description, descriptionText: "Edit address, select friends, select selenium's avatar, and select selenium's favorite foods.", hash: "4208"}
2:{plan: Recipe, rank: 0.000012800000000000008, description: Description, descriptionText: "Undefined.", hash: "b658"}
3:{plan: Recipe, rank: 0.00009216000000000005, description: Description, descriptionText: "Show products from your browsing context (Minecraf…er items) (Book: How to Draw plus 2 other items).", hash: "a0de"}
4:{plan: Recipe, rank: 0.00009216000000000005, description: Description, descriptionText: "Show products from your browsing context (Minecraf…er items) (Book: How to Draw plus 2 other items).", hash: "30ec"}
```

**Comment by shaper** [comment]
@mmandlis this looks like it could be a description bug though it could also have been introduced by recent app shell changes around triggering and throttling suggestions. But I wouldn't think we'd ever want to have runtime return an `Undefined` `descriptionText` so figure you would want to look at this in any case.

**Comment by seefeldb** [comment]
I was going to update those descriptions to Maria's new method anyway, so I assigned it to me as well.

**Comment by mariakleiner** [comment]
Currently it is happening because the particle's description is set to
description `make a reservation: ${event.description}`
and unless a restaurant is selected, the `event.description` is undefined, which turns the whole description to undefined.

The change in arcs-cdn that caused the tests to start failing is that previously after a restaurant was selected, the suggestions were regenerated, and with a selected restaurant, there would be a valid suggestion. Now they aren't regenerated.

**Comment by shaper** [comment]
OK, I'm sending a tiny PR to mark the one test to `skip` for now so we have a green build and will see any other test failures in meantime.

**Comment by smalls** [comment]
I think @seefeldb fixed this, maybe as part of 3dfb793a65fa27ce641a555fdd5ecca72ca5861e. I'll close this in the hopes that the test is less flaky.
